app/components/identity-steps/step-four.js

```diff
@@ -2,26 +2,65 @@ import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
 import { inject as service } from '@ember/service';
+import { APPS } from '../../constants/urls';
+import checkURL from '../../utils/check-url';
 import { TOAST_OPTIONS } from '../../constants/toast-options';
 
-export default class IdentityStepsStepFourComponent extends Component {
+export default class IdentityStepsStepFiveComponent extends Component {
   @service toast;
-  @tracked hideChaincode = true;
-  @tracked isCopyClicked = false;
-  @tracked isChaincodePageButtonDisabled = true;
-  @tracked isTooltipVisible = true;
+  @tracked isMouseOnTooltip = false;
+  @tracked profileURL = '';
+  @tracked nextButtonDisabled = true;
+  @tracked isLoading = false;
 
-  @action toggleEye() {
-    this.hideChaincode = !this.hideChaincode;
+  @action openTooltipInfo() {
+    this.isMouseOnTooltip = true;
   }
 
-  @action handleCopy() {
-    navigator.clipboard.writeText(this.Chaincode);
-    this.isCopyClicked = true;
-    this.isTooltipVisible = false;
-    this.isChaincodePageButtonDisabled = false;
-    if (this.isCopyClicked === true) {
-      this.toast.info('Copied', '', TOAST_OPTIONS);
+  @action closeTooltipInfo() {
+    this.isMouseOnTooltip = false;
+  }
+
+  @action changeProfileURL(e) {
+    this.profileURL = e.target.value;
+    if (this.profileURL === '' || !checkURL(this.profileURL)) {
+      this.nextButtonDisabled = true;
+    } else {
+      this.nextButtonDisabled = false;
+    }
+  }
+
+  @action async handleEdit(e) {
+    e.preventDefault();
+    this.isLoading = true;
+    try {
+      const response = await fetch(`${APPS.API_BACKEND}/users/profileURL`, {
+        method: 'PATCH',
+        body: JSON.stringify({ profileURL: this.profileURL }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'include',
+      });
+      if (response.ok) {
+        this.toast.info('Profile Service Updated!!', 'Info', TOAST_OPTIONS);
+      } else {
+        this.toast.error(
+          'Something went wrong. Please check console errors.',
+          'Error',
+          TOAST_OPTIONS,
+        );
+      }
+    } catch (error) {
+      console.error(error);
+      this.toast.error(
+        'Something went wrong. Please check console errors.',
+        'Error',
+        TOAST_OPTIONS,
+      );
+    } finally {
+      this.isLoading = false;
+      this.args.startHandler();
```

Comment:
why moving to next step in finally block, this block will run either request is success or failure